robyn/router.py

```diff
@@ -336,11 +339,12 @@ def inner_handler(*args, **kwargs):
                     self.add_route(
                         middleware_type,
                         endpoint,
+                        None,
                         async_inner_handler,
                         injected_dependencies,
                     )
                 else:
-                    self.add_route(middleware_type, endpoint, inner_handler, injected_dependencies)
+                    self.add_route(middleware_type, endpoint, None, inner_handler, injected_dependencies)
```

Comment:
I am not super happy with the design here. 

Why do we need to pass a `None` in a subrouter add_route.

This needs a better design @VishnuSanal